src/main/java/com/github/teamfusion/spyglassplus/common/entity/SpyglassStandEntity.java

```diff
@@ -190,6 +221,24 @@ public ItemStack getSpyGlass() {
 		return this.entityData.get(DATA_SPYGLASS);
 	}
 
+	public void setOwner(Entity entity) {
+		if (entity == null) {
+			this.entityData.set(DATA_OWNER_ID, OptionalInt.empty());
+		} else {
+			this.entityData.set(DATA_OWNER_ID, OptionalInt.of(entity.getId()));
+		}
```

Comment:
If the owner is going to be a player, you can use the SynchedEntityData<Optional<UUID>> type and use the player's UUID, which might be better in this situation. If it's not always going to be a player, then this should work fine.